### ISSUE TITLE
SetInterfaceInfo picks up the wrong interface

### ISSUE THREAD

**Issue by yilongli** [eunyoung14/mtcp#150]
My NIC has two interfaces: ```eno1``` and ```eno1d1```.

The following buggy line will add ```eno1``` to the attached devices even though I have only enabled ```eno1d1``` in the config file.
https://github.com/eunyoung14/mtcp/blob/6babf3c75aa73a0c90c005d07e7de27ab6c135ab/mtcp/src/io_module.c#L230

**Comment by ajamshed** [comment]
Thanks yilongli.
I will look into this issue. Are you using Mellanox devices?

**Comment by yilongli** [comment]
Yes, I am using Mellanox ConnectX-3 10 GB NIC and [Mellanox DPDK](http://www.mellanox.com/page/products_dyn?product_family=209&mtag=pmd_for_dpdk).

**Comment by zorrohahaha** [comment]
Does this support Mellanox ConnectX-4 & DPDK mode?

**Comment by ajamshed** [comment]
It should work. Please note that you don't need to insert either ``igb_uio.ko`` or ``dpdk_iface.ko`` kernel drivers. I should probably update the README file to explain this. I will give you a step-by-step procedure below. Please follow these guidelines:

1- Open up ``setup_mtcp_dpdk_env.sh`` and comment (disable) lines 54-63.
2- Next follow the README file.
3- Instead of using ``dpdk0`` interface, you should use the actual name of the Ethernet interface as seen from ``ifconfig`` command.
4- Finally, you should replace the mTCP configuration (e.g. ``epserver.conf``) lines ``port = dpdk0`` to ``port = <intf_name>`` where ``<intf_name>`` is the actual name of the Ethernet interface as seen from ``ifconfig`` command.  You should do the same thing for the line: ``stat_print``.

Please let me know if you have any other follow-up questions. I will update the file to include these guidelines soon.
